lib/filterx/func-unset-empties.c

```diff
@@ -46,29 +66,54 @@ static gboolean _process_list(FilterXFunctionUnsetEmpties *self, FilterXObject *
 static gboolean
 _should_unset(FilterXFunctionUnsetEmpties *self, FilterXObject *obj)
 {
-  gsize str_len;
-  const gchar *str;
-  if (filterx_object_extract_string(obj, &str, &str_len))
+  if (check_flag(self->flags, FILTERX_FUNC_UNSET_EMPTIES_FLAG_REPLACE_NULL) &&
+      filterx_object_is_type(obj, &FILTERX_TYPE_NAME(null)))
+    return TRUE;
+
+  if ((check_flag(self->flags, FILTERX_FUNC_UNSET_EMPTIES_FLAG_REPLACE_EMPTY_LIST) &&
+       filterx_object_is_type(obj, &FILTERX_TYPE_NAME(list))) ||
+      (check_flag(self->flags, FILTERX_FUNC_UNSET_EMPTIES_FLAG_REPLACE_EMPTY_DICT) &&
+       filterx_object_is_type(obj, &FILTERX_TYPE_NAME(dict))))
     {
-      return str_len == 0 ||
-             strcasecmp(str, "n/a") == 0 ||
-             strcmp(str, "-") == 0;
+      guint64 len;
+      filterx_object_len(obj, &len);
+      return len == 0;
     }
 
-  if (filterx_object_is_type(obj, &FILTERX_TYPE_NAME(null)))
+  gsize str_len = 0;
```

Comment:
[optional]

The string part got quite large, it could go to a separate function for readability.